admin/bin/qserv-configure.py

```diff
@@ -1,311 +1,381 @@
-                sys.exit(1)
+        parser.add_argument('-v', '--verbose', dest='verbose', default=[],
+                            action='append_const',
+                            const=None,
+                            help='More verbose output, can use several times.')
+        parser = lsst.qserv.admin.logger.add_logfile_opt(parser)
+
+        parser.add_argument("-a", "--all", dest="all", action='store_true',
+                            default=False,
+                            help='''clean execution directory and then run all configuration steps'''
+                            )
+        # Defining option of each configuration step
+        for step_name in configure.STEP_LIST:
+            parser.add_argument(
+                "-{0}".format(configure.STEP_ABBR[step_name]),
+                "--{0}".format(step_name),
+                dest="step_list",
+                action='append_const',
+                const=step_name,
+                help=configure.STEP_DOC[step_name]
+            )
 
-        in_config_dir = os.path.join(qserv_dir, "cfg")
-        in_template_config_dir = os.path.join(in_config_dir, "templates")
-        out_template_config_dir = os.path.join(args.qserv_run_dir, "templates")
-        logging.info("Copying template configuration from {0} to {1}"
-                     .format(in_template_config_dir, args.qserv_run_dir)
-                     )
-        shutil.copytree(in_template_config_dir, out_template_config_dir)
-
-        in_meta_config_file = os.path.join(in_config_dir, "qserv-meta.conf")
-        logging.info("Creating meta-configuration file: {0}"
-                     .format(args.meta_config_file)
-                     )
-        params_dict = {
-            'RUN_BASE_DIR': args.qserv_run_dir
-        }
-        configure.apply_tpl_once(
-            in_meta_config_file, args.meta_config_file, params_dict)
+        # forcing options which may ask user confirmation
+        parser.add_argument("-f", "--force", dest="force", action='store_true',
+                            default=False,
+                            help="forcing removal of existing execution data"
+                            )
+
+        # run dir, all mutable data related to a qserv running instance are
+        # located here
+        parser.add_argument("-R", "--qserv-run-dir", dest="qserv_run_dir",
+                            default=default_qserv_run_dir,
+                            help="full path to qserv_run_dir"
+                            )
+
+        # meta-configuration file whose parameters will be dispatched in Qserv
+        # services configuration files
+        self.args = parser.parse_args()
+        default_meta_config_file = os.path.join(
+            self.args.qserv_run_dir, "qserv-meta.conf")
+        parser.add_argument("-m", "--metaconfig", dest="meta_config_file",
+                            default=default_meta_config_file,
```

Comment:
Adding new arguments after `parse_args()` is confusing. `qserv-configure.py -h` does not even show `-m` option.